subgit/core.py

```diff
@@ -486,6 +487,111 @@ def pull(self, names):
                 log.info(f"Current commit summary on HEAD in git repo '{name}': ")
                 log.info(f"  {str(repo.head.commit.summary)}")
 
+    def delete_local_repo(self, repo_names=None):
+        """
+        Method takes zero or one repo name as argument.
+
+        Checks if the repo(s) are valid git repo(s) and if there are any untracked changes.
+        """
+
+        config = self._get_config_file()
+
+        active_repos = self._get_active_repos(config)
+
+        if repo_names is None:  # If no names are specified in the command
+            active_repos_dict = config.get('repos', [])
+
+            repo_choices = ", ".join(active_repos_dict)
+
+            answer = self.yes_no(f"Are you sure you want to delete the following repos '{repo_choices}'")
+
+            if answer:
+                self.repo_deletion(active_repos=active_repos)
+
+        if isinstance(repo_names, list):  # If at least one name was specified
+
+            repo_choices = ", ".join(repo_names)
+
+            answer = self.yes_no(f"Are you sure you want to delete the following repos '{repo_choices}'")
+
+            if answer:
+                self.repo_deletion(active_repos, repo_names)
```

Comment:
So here you got a big code duplication. I would suggest that you move up the two if cases, create the correct data that you want to send through the otherwise shared code removing the unnecessary code duplication